### PR TITLE
do not blindly overwrite kube config

### PR DESCRIPTION
**Pull request by roivanov**
#### Changes ####

Merge with existing kube config, do not overwrite existing file

#### Linked Issues ####

#259 

One issue still needs to be addressed is when cluster names are identical

### CODE REVIEW

File: roles/k3s_server/tasks/main.yml [hunk 1]
```diff
@@ -93,9 +93,30 @@
...
         flat: true
 
+    - name: Check existing kube config
+      stat:
+        path: ~/.kube/config
+      register: result
+      delegate_to: 127.0.0.1
+      become: false
+
+    - name: Show results
```
Comment (roivanov) [inline roles/k3s_server/tasks/main.yml:106]: remove debug

File: roles/k3s_server/tasks/main.yml [hunk 2]
```diff
@@ -93,9 +93,26 @@
...
         flat: true
 
+    - name: Check existing kube config
+      ansible.builtin.stat:
+        path: ~/.kube/config
+      register: result
+      delegate_to: 127.0.0.1
+      become: false
+
+    - name: Merge with existing kube config
```
Comment (roivanov) [inline roles/k3s_server/tasks/main.yml:106]: FIXME cover case when file does not exist

File: roles/k3s_server/tasks/main.yml [hunk 3]
```diff
@@ -93,9 +93,47 @@
...
+        TFILE=$(mktemp)
+        kubectl config view --flatten > ${TFILE}
+        mv ${TFILE} ~/.kube/config
+        rm ~/.kube/config.new
+      delegate_to: 127.0.0.1
+      when: result.stat.exists
+      become: false
+      register: mv_result
+      changed_when:
+        - mv_result.rc == 0
```
[the remark this answers is not in the document]
Comment (dereknola) [reply roles/k3s_server/tasks/main.yml:118]: We don't need to check if an existing config exists, `kubectl` can handle a nonexistent file, you just end up with a copy of `config.new`, which is fine.

**Review by dereknola — CHANGES_REQUESTED** [review]

File: roles/k3s_server/tasks/main.yml [hunk 4]
```diff
@@ -93,9 +93,47 @@
...
+      changed_when:
+        - cp_result.rc == 0
+
+    - name: Clear access mode bits for kube config
+      ansible.builtin.file:
+        path: ~/.kube/config
+        state: file
+        mode: go=
+      delegate_to: 127.0.0.1
+      become: false
```
[the remark this answers is not in the document]
Comment (roivanov) [reply roles/k3s_server/tasks/main.yml]: tasks removed and tested locally

**Comment by zen** [comment]
Guys, why not allow user to define kubeconfig location? We use multiple configs and export KUBECONFIG based on project - it's very convenient

**Comment by dereknola** [comment]
@zen I'm happy to open another future PR to address that case. I did think it was weird that this project originally overrode the base kubeconfig, but I kept it in because I didn't know if users actually liked that or what.

**Pull request merged**